tests/ui/editions/nested-macro-rules-edition-local-2024.rs

```diff
@@ -0,0 +1,20 @@
+// See nested-macro-rules-edition-local-2021.rs for a description of this test.
+//
+// This flips the editions to show the opposite behavior.
+
+//@ aux-crate: nested_macro_rules_dep_2021=nested_macro_rules_dep_2021.rs
+//@ edition:2024
+//@ check-pass
```

Comment:
Question: could this be coalesced into the `-2021` test to become a test with two differential revisions, with the "root" test crate being revisioned on edition 2021 vs 2024 to more obviously show the difference in behavior?

Reply:
Sounds good. Though I can't think of a way to extend that to the aux-crate, which is only differs in the edition.

Reply:
> Though I can't think of a way to extend that to the aux-crate, which is only differs in the edition.

In theory they might be able to `include!` some common file and just differ in the compile-flags, but I don't forsee us changing these two aux files often anyways, so seems overkill. The current duplication seems completely fine.